### PR TITLE
fix 'or' resolver

### PR DESCRIPTION
**Pull request by klymenko-serhii**
if wrapped component thrown an error, this error should be returned
cover local error case with unit test

ref: https://github.com/thebigredgeek/apollo-resolvers/issues/46

### CODE REVIEW

**Comment by newsiberian** [comment]
@thebigredgeek , hi, could you, please, review this PR? my tests broken because of this, can't do deploy things

**Comment by newsiberian** [comment]
Looking deeper at this change, I understand that it is invalid. It broke logic to other cases rather than "true, false, throw". It returns validation resolvers result instead of main resolver result.

File: src/helper.ts [hunk 1]
```diff
@@ -16,12 +16,16 @@ export const and = (...conditions) => resolver => {
...
-export const or = (...conditions) => resolver => (...query) => {
+export const or = (...conditions) => resolver => (root, args, context, info) => {
   return new Promise((resolve, reject) => {
     let limit = conditions.length - 1;
     const attempt = (i) =>
-      conditions[i].createResolver(resolver)(...query)
-        .then(res => resolve(res))
+      createResolver(conditions[i])(root, args, context, info)
+        .then(() => {
+          createResolver(resolver)(root, args, context, info)
```
Comment (newsiberian) [inline src/helper.ts]:
```ts
return createResolver(resolver)(root, args, context, info)
```

**Comment by thebigredgeek** [comment]
Would love to either get this merged or close it out @klymenko-serhii  .  Can we get tests passed?

**Comment by thebigredgeek** [comment]
Released with https://github.com/boltsource/apollo-resolvers/tree/1.4.2

**Pull request merged**